addons/sale/models/sale.py

```diff
@@ -182,7 +182,7 @@ def _search_invoice_ids(self, operator, value):
 
     user_id = fields.Many2one(
         'res.users', string='Salesperson', index=True, tracking=2, default=lambda self: self.env.user,
-        domain=lambda self: [('groups_id', 'in', self.env.ref('sales_team.group_sale_salesman').id)])
+        domain=lambda self: [('groups_id', 'in', self.env.ref('sales_team.group_sale_salesman').id), ('share', '=', False), ('company_id', '=', self.env.company.id), (self.env.user.has_group('sales_team.group_sale_salesman'), '=', True)])
```

Comment:
Why the `(self.env.user.has_group('sales_team.group_sale_salesman'), '=', True)` ?

Reply:
I thought I had to add something to avoid showing non salesman user but it was already in the domain. My bad